internal/model/labels.go

```diff
@@ -59,6 +61,53 @@ func (l Labels) Clone() Labels {
 	return cp
 }
 
+// Equal returns true when the two maps of labels are equal.
+func (l Labels) Equal(labels Labels) bool {
+	for key, localV := range l {
+		v, ok := labels[key]
+		if !ok {
+			return false
+		}
+		if v.Value != localV.Value {
+			return false
+		}
+		if len(v.Values) != len(localV.Values) {
+			return false
+		}
+		for _, s := range [][]string{v.Values, localV.Values} {
+			if !sort.StringsAreSorted(s) {
+				sort.Strings(s)
+			}
+		}
```

Comment:
Seems dangerous and wrong to me for an `Equal` method to mutate the value. Do we need to sort them?